lib/hooks/userspace/hook-koa.js

```diff
@@ -0,0 +1,130 @@
+/**
```

Comment:
I've added some `consts` throughout. If there are particular features you have in mind I can definitely incorporate them.